src/App.js

```diff
@@ -273,20 +284,128 @@ class App extends React.Component<PropsT, StateT> {
       displayed = (
         <GameStart
           onClickStart={() => {
-            this.setState({
-              hasBeenWelcomed: true,
-              view: "training"
-            });
-            if (isRegistered && userId) {
-              this.recordNewSession(userId);
-            }
+            this.gameStartButtons("parallelograms", isRegistered, userId)
+          }}
+          onClickStartMamiferes={() => {
+            this.gameStartButtons("mammals", isRegistered, userId)
+          }}
+          onClickStartAdverbs={() => {
+            this.gameStartButtons("adverbs", isRegistered, userId)
           }}
           studentName={this.studentName}
           studentImg={this.studentBackpackImg}
           genderTeacherMale={this.genderTeacherMale}
         />
       );
-    } else if (view === "leaderboard") {
+    } else if( view === "adverbs"){
+      displayed = <Adverbs />;
+
+    } else if (view === "mammals"){
```

Comment:
The 100 following lines are duplicated with the condition `if (view === "parallelograms" ) { ... }`. I suggest that you can simply go back to using `view === " training"`, but add a new state value `topic = "mammals" | "parallelograms" | "adverbs"`. Then you pass the topic as a property to the following components which display the correct course.